src/hooks/useMemelist.ts

```diff
@@ -0,0 +1,49 @@
+'use client';
+import { useQuery } from '@tanstack/react-query';
+import { StrapiTokenInfo } from 'src/types/tokenList';
+
+export interface UseMemelistProps {
+  tokens?: StrapiTokenInfo[];
+  isSuccess: boolean;
+}
+
+export interface UseMemeProps {
+  enabled: boolean;
+}
+
+const STRAPI_CONTENT_TYPE = 'token-lists';
+export const useMemelist = ({ enabled }: UseMemeProps): UseMemelistProps => {
+  const apiBaseUrl =
+    process.env.NEXT_PUBLIC_STRAPI_DEVELOP === 'true'
```

Comment:
Would be good to have a function to load the right strapi endpoint to have a better codebase + avoid any mistake

e.g.
```js
function getPublicStrapiUrl() {
  return process.env.NEXT_PUBLIC_STRAPI_DEVELOP === 'true'
  ? process.env.NEXT_PUBLIC_LOCAL_STRAPI_URL
      : process.env.NEXT_PUBLIC_STRAPI_URL;
}
```